pkg/bootstrap/platform/gcp.go

```diff
@@ -52,8 +54,12 @@ var (
 	GCPQuotaProjectVar = env.RegisterStringVar("GCP_QUOTA_PROJECT", "", "Allows specification of a quota project to be used in requests to GCP APIs.").Get()
 )
 
+var isMetadataEndpointAccessible bool
+
 var (
-	shouldFillMetadata = metadata.OnGCE
+	shouldFillMetadata = func() bool {
```

Comment:
Do we want just a function defined here? it seems we end up calling this quite a bit.. does it make sense to store the result in the `gcpEnv` struct?